js/controller.js

```diff
@@ -993,8 +1000,46 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
         clearTimeout(this.state.timer);
         this.state.timer = null;
       }
+    },
+
+    //determine how aspect ratio is enabled
+    enableAspectRatio: function() {
+      //use fixed aspect ratio number from skinConfig
+      if(this.skin.props.skinConfig.responsive.aspectRatio && this.skin.props.skinConfig.responsive.aspectRatio != "auto") {
+        this.state.mainVideoAspectRatio = this.skin.props.skinConfig.responsive.aspectRatio;
+        this.setAspectRatio(this.state)
+      }
+      //auto detect and update aspect ratio (default)
+      else {
+        if(this.state.isInitialPlay) {
```

Comment:
isInitialPlay flag is for 'auto detect' only. If the aspect ratio is set in skinConfig it can be set immediately for startScreen before the video is played and metadata loaded.